server/bastionlab_polars/src/composite_plan.rs

```diff
@@ -24,7 +25,7 @@ pub enum CompositePlanSegment {
 }
 
 impl CompositePlan {
-    pub fn run(self, state: &BastionLabPolars) -> Result<DataFrameArtifact, Status> {
+    pub fn run(self, state: &BastionLabPolars, token: Option<Bytes>) -> Result<DataFrameArtifact, Status> {
```

Comment:
Maybe pass a Context object here which already contains the session info?
Then this function only fills the aggregation info and passes it to the policy checker?

Reply:
Done.